protos/flyteidl/core/tasks.proto

```diff
@@ -95,6 +103,9 @@ message TaskMetadata {
 
     // Indicates whether the system should attempt to execute discoverable instances in serial to avoid duplicate work
     bool cache_serializable = 9;
+
+    // Extra metadata about the deck
+    DeckMetadata deck = 10;
```

Comment:
we cannot add this here. It is completely dynamic for every task execution. It should be like an output